data-prepper-plugins/aggregate-processor/src/test/java/com/amazon/dataprepper/plugins/processor/aggregate/actions/RemoveDuplicatesAggregateActionTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate.actions;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateAction;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class RemoveDuplicatesAggregateActionTest {
```

Comment:
Are we missing the test which demonstrates receiving an event which is already in the group state does not update the group state?

Do I have a correct understanding of the remove_duplicates expected functionality?

Reply:
I can make a test for this, but the groupState is never modified directly by `remove_duplicates` even for the first one where the groupState doesn't contain anything. The logic of modifying the groupState for the first Event that is a part of that groupState will be handled by the `AggregateProcessor`. So the flow will be somewhat like this.

In AggregateProcessor:
* check if the groupState exists for a certain Event (if it doesn't then groupState is an empty `Map<Object, Object>)`
* Call `aggregateAction.handleEvent(groupState)`
* Add the starting Event to the `groupState`

With this logic RemoveDuplicates will never modify the groupState no matter what.

Reply:
However, there is a consideration to let `AggregateAction` creators handle this logic. It would just be duplicated for pretty much all AggregateActions